docker/BaseDockerContainer/scripts/gridEngineSetup.sh

```diff
@@ -83,37 +83,40 @@ function installGridEngine() {
     qconf -am roddy
     qconf -au roddy users
     qconf -as $HOST
+	# define roddy as the user to run SGE (required by Singularity)
+	sed -i -e 's/admin_user.*/admin_user roddy/' /etc/gridengine/bootstrap /var/lib/gridengine/default/common/bootstrap
+	# make SGE-related files readable by everyone so that any user can run SGE
+	chmod -R a+rx /var/spool/gridengine
 }
 
-function setupGridEngine() {
-cat >/tmp/qconf-editor.sh <<EOF
-#!/bin/sh
-sleep 1
-perl -pi -e 's/^hostname.*$/hostname $hostName/' \$1
-EOF
-
-chmod +x /tmp/qconf-editor.sh
-export EDITOR=/tmp/qconf-editor.sh
-qconf -ae
-cat >/tmp/qconf-editor.sh <<EOF
-#!/bin/sh
+# some parameters of gridengine can only be modified using an interactive editor
+# this function creates a temporary non-interactive script that mimicks an interactive editor
+# Usage: changeGridEngineConfig PARAMETER_TO_CHANGE NEW_VALUE QCONF_ARGUMENTS
+function changeGridEngineConfig() {
+PARAMETER="$1"
```

Comment:
Note: You can avoid creating global variables using `local`

```bash
     local PARAMETER="$1"
```

Also consider checking correct function usage

```bash
    local PARAMETER="${1:?PARAMETER no defined}"
```

Both will improve you code stability for this function that you use multiple times.